tests/test-runner/cmd/test-runner.py

```diff
@@ -874,8 +879,7 @@ def main(args):
 
     testrun.complete_outputdirs(options)
     testrun.run(options)
-    testrun.summary()
-    exit(0)
+    exit(testrun.summary())
```

Comment:
Rather than do this unconditionally let's add a command line option to control this behavior.  Tools like `valgrind` and `cppcheck` provide a `--error-exitcode` to control the exit code on error.